git_machete/git_operations.py

```diff
@@ -911,6 +958,22 @@ def get_commits_between(self, earliest_exclusive: AnyRevision, latest_inclusive:
             utils.get_non_empty_lines(self._popen_git("log", "--format=%H:%h:%s", f"^{earliest_exclusive}", latest_inclusive, "--").stdout)
         ))))
 
+    def get_diffs_between(self, earliest_exclusive: AnyRevision, latest_inclusive: AnyRevision) -> Dict[FullCommitHash, str]:
+        # Returns a dictionary of git hashes and their diffs between two revisions
+        marker = "__MACHETE___COMMIT_MARKER__"  # Use a marker so we can split the output into individual diffs for each commit
+        out = self._popen_git("log", "--patch", f"--format=format:{marker}%H", f"^{earliest_exclusive}", latest_inclusive, "--").stdout
```

Comment:
And indeed, maybe add some limit on number of commits included in the results... like `-1000` option (1000 extracted to `constants.py`)

Reply:
Done! (sorry about the delays in getting to your comments, had a busy week :D)

Reply:
Thanks! I'll take a look once I'm done with my commercial work ⚒️